app/src/test/java/com/example/marcinwisniewski/intivetrainingapp/RxImmediateSchedulerRule.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.marcinwisniewski.intivetrainingapp
+
+import io.reactivex.Scheduler
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.disposables.Disposable
+import io.reactivex.internal.schedulers.ExecutorScheduler
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+
+class RxImmediateSchedulerRule : TestRule {
+    private val immediate = object : Scheduler() {
+        override fun scheduleDirect(run: Runnable, delay: Long, unit: TimeUnit): Disposable {
+            return super.scheduleDirect(run, 0, unit)
+        }
+
+        override fun createWorker(): Scheduler.Worker {
+            return ExecutorScheduler.ExecutorWorker(Executor { it.run() })
+        }
+    }
+
+    override fun apply(base: Statement, description: Description): Statement {
+        return object : Statement() {
+            @Throws(Throwable::class)
+            override fun evaluate() {
+                RxJavaPlugins.setInitIoSchedulerHandler { immediate }
```

Comment:
I am not sure about this part of code - I am still reading about how it works. I could easily change immediate to Schedulers.trampoiline() in initMainThreadScheduler but with other I had some problems - I am not sure if I need more than setInitIoSchedulerHandler and setInitMainThreadSchedulerHandler in this evaluate method.